pkgs/development/libraries/rocsolver/default.nix

```diff
@@ -0,0 +1,104 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, writeScript
+, cmake
+, rocm-cmake
+, rocm-runtime
+, rocm-device-libs
+, rocm-comgr
+, rocblas
+, hip
+, fmt
+, gtest ? null
+, gbenchmark ? null
+, buildTests ? false
+, buildBenchmarks ? false
+, gpuTargets ? null # gpuTargets = [ "gfx803" "gfx900" "gfx906:xnack-" ]
+}:
+
+assert buildTests -> gtest != null;
+assert buildBenchmarks -> gbenchmark != null;
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "rocsolver";
+  repoVersion = "3.19.0";
+  rocmVersion = "5.3.3";
+  version = "${finalAttrs.repoVersion}-${finalAttrs.rocmVersion}";
+
+  outputs = [
+    "out"
+  ] ++ lib.optionals buildTests [
+    "test"
+  ] ++ lib.optionals buildBenchmarks [
+    "benchmark"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "ROCmSoftwarePlatform";
+    repo = "rocSOLVER";
+    rev = "rocm-${finalAttrs.rocmVersion}";
```

Comment:
Then version should be equal to rocmVersion

Reply:
Really!? That tag should be 3.19.0...

![image](https://user-images.githubusercontent.com/7258858/204158350-be7cc6ac-a778-4616-ad3d-2df1f14c3757.png)

Reply:
Either that just changed I had a major derp. I'll fix it.

Reply:
Oh, you commented on something else, my bad.
The rocm (repo) releases seem to differ from the actual tags.
From what I understand, the repo has it's own version, and it updated to each rocm version.
The repo version seems to be repo-specific changes.